File: Programmers/Algorithm_Kit/05_brute_force/002.cpp
#include <string>
#include <vector>
#include <algorithm> //max_element
#include <iostream>
using namespace std;

int cnt[3];

vector<int> solution(vector<int> answers) {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);

    vector<int> answer;
    
    vector<vector<int>> list = {
        {1, 2, 3, 4, 5},
        {2, 1, 2, 3, 2, 4, 2, 5},
        {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}
    };
    
    for (int i = 0; i < answers.size(); i++) {
        if (answers[i] == list[0][i % list[0].size()]) cnt[0]++;
        if (answers[i] == list[1][i % list[1].size()]) cnt[1]++;
        if (answers[i] == list[2][i % list[2].size()]) cnt[2]++;
    }
    
    int max_cnt = *max_element(cnt, cnt + 3);
    
    for (int i = 0; i < 3; i++) {
        if (cnt[i] == max_cnt) {
            answer.push_back(i + 1);
        }
    }
    
    return answer;
}

File: programmers/algorithmkit/10_graph/002.java
class Solution {
    public int solution(int n, int[][] results) {
        int answer = 0;

        int[][] adj = new int[n + 1][n + 1];

        for (int[] result : results) {
            adj[result[0]][result[1]] = 1;
            adj[result[1]][result[0]] = -1;
        }

        // 플로이드 워셜
        for (int k = 1; k <= n; k++) {
            for (int i = 1; i <= n; i++) {
                for (int j = 1; j <= n; j++) {
                    if (adj[i][k] == 1 && adj[k][j] == 1) {
                        adj[i][j] = 1;
                        adj[j][i] = -1;
                    }
                }
            }
        }

        for (int i = 1; i <= n; i++) {
            int cnt = 0;

            for (int j = 1; j <= n; j++) {
                if (adj[i][j] != 0) {
                    cnt++;
                }
            }

            if (cnt == n - 1) { // i가 모든 j에 대해 결과가 정해지는 경우
                answer++;
            }
        }

        return answer;
    }
}
